src/kvstore/kvstore_dist_server.h

```diff
@@ -170,6 +187,33 @@ class KVStoreDistServer {
     app->Response(recved);
   }
 
+  void SetProfilerConfig(std::string params_str) {
+    std::vector<std::string> elems;
+    mxnet::kvstore::split(params_str, ',', std::back_inserter(elems));
+    std::vector<const char*> ckeys;
+    std::vector<const char*> cvals;
+    ckeys.reserve(elems.size());
+    cvals.reserve(elems.size());
+
+    for (int i=0; i < elems.size(); i++) {
+      std::vector<std::string> parts;
+      mxnet::kvstore::split(elems[i], ':', std::back_inserter(parts));
+      CHECK_NOTNULL(parts[0].c_str());
+      CHECK_NOTNULL(parts[1].c_str());
+      if (parts[0] == "filename") {
+        parts[1] = "rank" + std::to_string(ps::MyRank()) + "_" + parts[1];
```

Comment:
c_str() will never return null

Reply:
You're right, not sure why I chose to do this. I'll remove the checks

Reply:
Actually let me add check that the parts are not empty

Reply:
You're right. Removed the checks